linux/deploy_vm/rebuild_unattend_install_iso.yml

```diff
@@ -21,245 +21,254 @@
+          - src_file: "{{ src_iso_file_dir }}/isolinux.cfg"
+            dest_file: "isolinux/isolinux.cfg"
+          - src_file: "{{ src_iso_file_dir }}/gtk.cfg"
+            dest_file: "isolinux/gtk.cfg"
+          - src_file: "{{ src_iso_file_dir }}/md5sum.txt"
+            dest_file: "md5sum.txt"
+          - src_file: "{{ new_unattend_install_conf }}"
+            dest_file: "{{ unattend_install_file_name }}"
+  when: unattend_install_conf is match('Debian')
 
-        - name: "Update timeout for boot menu"
-          ansible.builtin.replace:
-            path: "{{ unattend_iso_cache }}/isolinux.cfg"
-            regexp: "default .*"
-            replace: "default installgui"
+- block:
+    - name: Extract specific files inside ISO
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - 'boot/grub/grub.cfg'
+          - 'md5sum.txt'
+    
+    - name: "Add autoinstall to UEFI boot kernel command for Ubuntu live server"
+      ansible.builtin.replace:
+        path: "{{ unattend_iso_cache }}/grub.cfg"
+        regexp: '(.*vmlinuz)(.*)'
+        replace: "\\1 autoinstall \\2"
 
-        - name: "Update Debian grub.cfg for autoinstall"
-          ansible.builtin.blockinfile:
-            path: "{{ unattend_iso_cache }}/grub.cfg"
-            block: |
-              set default="autoinstall"
-              set timeout=5
-              menuentry "Automated installation" --id autoinstall {
-                  set background_color=black
-                  linux    /{{ debian_install_type }}/vmlinuz auto=true file=/cdrom/{{ unattend_install_file_name }} vga=788 --- quiet
-                  initrd   /{{ debian_install_type }}/gtk/initrd.gz
-              }
+    - name: "Set timeout to 5 seconds at boot menu"
+      ansible.builtin.replace:
+        path: "{{ unattend_iso_cache }}/grub.cfg"
+        regexp: 'set timeout=.*'
+        replace: "set timeout=5"
+
+    - name: "Update md5sum for UEFI boot config file"
+      ansible.builtin.shell: |
+        md5=`md5sum grub.cfg | awk '{print $1}'`
+        sed -i "/.\/boot\/grub\/grub.cfg/ s/^[^ ]*/$md5/" md5sum.txt
+      args:
+        chdir: "{{ unattend_iso_cache }}"
+    
+    - name: "set var ubuntu_bios_cfg_exist to default false" 
+      ansible.builtin.set_fact:
+        ubuntu_bios_cfg_exist: false
 
-        - name: "Update boot menu with preseed.cfg for Debian"
-          ansible.builtin.replace:
-            path: "{{ unattend_iso_cache }}/gtk.cfg"
-            regexp: '(.*)(initrd.gz )(.*)'
-            replace: "\\1\\2 auto=true file=/cdrom/{{ unattend_install_file_name }} \\3"
+    - name: Extract isolinux/txt.cfg inside ISO if exists
+      community.general.iso_extract:
+        image: "{{ src_iso_file_path }}"
+        dest: "{{ src_iso_file_dir }}"
+        files:
+          - 'isolinux/txt.cfg'
+      register: check_file_result
+      failed_when: false 
```

Comment:
Please use `ignore_errors: true` instead of `failed_when: false`

Reply:
Thanks. Done